Missing protein site

skhilton opened the pull request:
This PR handles corner cases involved in the missing data for the protein site/chain _and_ when there is only one line of data. 

1. missing protein chain / protein site
Before this PR, if the user tried to select a site with a _valid_ (not missing value) chain and a _valid_ (not missing value) site but the chain/site combo did not exist in the pdb then nothing happened. However, if either the protein chain or the protein site is a missing value then all of the sites were selected. Now, the code first checks that neither the protein site nor the protein chain are missing values before calling the `selectSiteOnProtein` function.

2. Weird color scheme with only one site
@jbloom's small test case revealed another, separate 🐛! The site plot color scheme is defined by the range, specifically at some point we _divide_ by the range. If you only have one site, the range is zero. And dividing by zero is undefined. Now, the code just checks if the range is zero and, if so, sets the range to 1.

I've attached the small test datasets I used to debug. The difference between `test1` and `test1000` is that test 1 includes a real value for protein site _and_ it exists on the protein structure. `test1000` includes a real value for the protein site _but_ the site does not exist on the protein structure. 

This resolves #169

skhilton committed:
handle null values for protein site

skhilton committed:
handle missing chains as well

huddlej reviewed (approved):
Thank you for figuring these out, @skhilton! I couldn't get any sites to appear on the protein view when the default flu data and these changes. I just pushed a commit that resolved the issue for me locally and confirmed that the code still works as expected when the protein site isn't defined.

Do you want to doublecheck that this works for you locally after my change?

huddlej committed:
Check for non-missing protein chain instead of attribute of chain

Corrects a logical test that could never evaluate to true and prevented any
sites from appearing selected on the protein view.

skhilton committed:
handle blank chains

skhilton commented:
@huddlej: I was having problems with the blank `protein_chain` test. How does this work for you?

huddlej commented:
I didn't test the missing chain, but I can see now that that bit was broken. :-x 

Your last commit fixes the missing chain case for me locally though. Thank you!
